config/packages/security.yaml

```
@@ -1,30 +1,43 @@
 security:
-    encoders:
-        App\Entity\Contractor:
-            algorithm: auto
+  encoders:
+    App\Entity\Contractor:
+      algorithm: auto
+    App\Entity\User:
+      algorithm: auto
 
-    # https://symfony.com/doc/current/security.html#where-do-users-come-from-user-providers
-    providers:
-        # used to reload user from session & other features (e.g. switch_user)
-        app_user_provider:
-            entity:
-                class: App\Entity\Contractor
-                property: username
-        # used to reload user from session & other features (e.g. switch_user)
-    firewalls:
-        dev:
-            pattern: ^/(_(profiler|wdt)|css|images|js)/
-            security: false
-        main:
-            anonymous: true
-            guard:
-                authenticators:
-                    - App\Security\ContractorAuthenticator
-            logout:
-                path: app_logout
-                invalidate_session: false
-                success_handler: App\Security\LogoutHandler
+  providers:
+    app_user_provider:
+      entity: { class: App\Entity\Contractor, property: username }
+    app_admin_provider:
+      entity: { class: App\Entity\User, property: email }
+  firewalls:
+    dev:
+      pattern: ^/(_(profiler|wdt)|css|images|js)/
+      security: false
+    admin:
+        pattern: ^/admin/
```

```suggestion
        pattern: ^/admin
```
Because when you enter /admin it is not redirecting you to admin/login